### PR TITLE
split tests

### PR DESCRIPTION
**Pull request by desistefanova**
Splitting tests methods into separated files.

This PR also fixes isReadOnly to use the correct Realm schema mode

We need to use "dart test -r expanded -j 1" since "-r expanded"  seems to have a bug and reports a single test name for all tests executed. this PR changes that as well

### CODE REVIEW

File: test/collections_test.dart [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+import 'test_base.dart';
+import '../lib/realm.dart';
+import 'test_model.dart';
+
+Future<void> main([List<String>? args]) async {
+  parseTestNameFromArguments(args);
+
+  print("Current PID $pid");
+
+  setupTests(Configuration.filesPath, (path) => {Configuration.defaultPath = path});
```
[the remark this answers is not in the document]
Comment (blagoev) [reply test/collections_test.dart:32]: all the config changes can happen inside `setupTests()`

[the remark this answers is not in the document]
Comment (desistefanova) [reply test/collections_test.dart:32]: ok

File: test/collections_test.dart [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+  test('Results notifications', () async {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/collections_test.dart:34]: OK

File: test/test_base.dart [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+// ignore_for_file: unused_local_variable
+
+import 'dart:io';
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/test_base.dart:21]: OK

File: test/config_test.dart [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+// ignore_for_file: unused_local_variable
+
+import 'dart:io';
+import 'package:test/test.dart' hide test, throws;
+import 'test_base.dart';
+import '../lib/realm.dart';
+import 'test_model.dart';
+
+Future<void> main([List<String>? args]) async {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/config_test.dart:27]: OK

File: test/list_test.dart [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+
+    var team = Team("TeamOne");
+    realm.write(() => realm.add(team));
+    var players = team.players;
+    realm.write(() => realm.delete(team));
+    expect(() => realm.write(() => realm.deleteMany(players)), throws<RealmException>("Access to invalidated Collection object"));
+    realm.close();
+  });
+
+  test('Realm deleteMany from list after realm is closed', () {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/list_test.dart]: OK

File: test/list_test.dart [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+
+    expect(teams, isNotNull);
+    expect(teams.length, 1);
+    final players = teams[0].players as RealmList<Person>;
+    expect(players.isValid, true);
+    realm.close();
+    expect(players.isValid, false);
+  });
+
+  test('Realm deleteMany from list referenced by two objects', () {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/list_test.dart]: OK

File: test/list_test.dart [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+    expect(teams[0].players[1].name, "Kimi");
+
+    expect(teams[0].scores.length, 3);
+    expect(teams[0].scores[0], 1);
+    expect(teams[0].scores[1], 2);
+    expect(teams[0].scores[2], 3);
+    realm.close();
+  });
+
+  test('Realm deleteMany from iterable', () {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/list_test.dart]: OK

File: test/list_test.dart [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+    //Ensure all players are deleted from collection
+    expect(teams[0].players.length, 0);
+
+    //Reload all persons from realm and ensure they are deleted
+    final allPersons = realm.all<Person>();
+    expect(allPersons.length, 0);
+    realm.close();
+  });
+
+  test('RealmObject add with list properties', () {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/list_test.dart]: Moved to realm_test.dart

File: test/list_test.dart [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+    //Delete teams one and three from realm
+    realm.write(() => realm.deleteMany([teamOne, teamThree]));
+
+    //Ensure both teams are deleted and only teamTwo has left
+    expect(teams.length, 1);
+    expect(teams[0].name, teamTwo.name);
+    realm.close();
+  });
+
+  test('Realm deleteMany from realm list', () {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/list_test.dart]: OK

File: test/realm_object_test.dart [hunk 10]
```diff
@@ -0,0 +1,428 @@
...
+    var realm = Realm(config);
+
+    realm.write(() {
+      realm.add(Car(''));
+    });
+
+    realm.close();
+  });
+
+  test('Realm add multiple objects', () {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/realm_object_test.dart]: OK

File: .github/workflows/realm-dart-linux.yml [hunk 11]
```diff
@@ -36,7 +36,7 @@ jobs:
 
       - name: Run tests
         run: |
-          dart test -r expanded --test-randomize-ordering-seed random
+          dart test -r expanded -j 1 --test-randomize-ordering-seed random
```
Comment (nielsenko) [inline .github/workflows/realm-dart-linux.yml:39]: Why do we need this? It prevents parallel execution of tests. Is it not possible that will catch some corner cases, even if it does introduce some non-determinism?

Comment (blagoev) [reply .github/workflows/realm-dart-linux.yml:39]: It is called out in the PR description. 
Our CI does not do parallel execution anyways it seems, I guess since it has a single CPU core per job.

**Review by nielsenko — APPROVED** [review]
